pkg/internal/ebpf/httpfltr/httpfltr.go

```diff
@@ -79,10 +80,27 @@ func New(cfg *pipe.Config, metrics imetrics.Reporter) *Tracer {
 }
 
 func (p *Tracer) AllowPID(pid uint32) {
+	if p.bpfObjects.ValidPids != nil {
```

Comment:
I piggybacked on the AllowPID and BlockPID to update the bpf map with the same information as the userspace code has it. However there's a small chicken and egg thing. When we create the tracer program we still haven't loaded it into the bpf space, so the map isn't created on the bpf side.

After we create the tracer for a given pid, we load it and then we call Run(). When Run() is called, I use the new CurrendPIDs helper, get the current state of the pids we had on boot of Beyla and I repeat the AllowPID process. This time the bpf map is created and we can set the pids we want in kernel space.

From then on, the AllowPID and BlockPID update both.